src/main/java/org/terasology/climateConditions/HyperthermiaSystem.java

```diff
@@ -73,4 +87,17 @@ public void onSpawn(OnPlayerSpawnedEvent event, EntityRef player, LocationCompon
         }
     }
 
+    private void weakenPlayer(EntityRef player, HealthComponent health) {
+        health.maxHealth *= healthReduceFactor;
+        health.currentHealth *= healthReduceFactor;
+        health.regenRate *= healthReduceFactor;
+        player.saveComponent(health);
+    }
+
+    private void removePlayerWeakness(EntityRef player, HealthComponent health) {   
+        health.maxHealth /= healthReduceFactor;
+        health.currentHealth /= healthReduceFactor;
```

Comment:
Same here - or rather: don't do anything. The current health should stay as is, and regen will recover over time.

Reply:
This, I agree to.

Reply:
No wait, this is weird. Either both need to change or neither one does.. 
Could you please read the above comment and tell me your thoughts, whether the current health should be scaled or not.